ballerina/tests/test.bal

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/io;
+
+configurable boolean isLiveServer = ?;
+configurable string token = ?;
+configurable string serviceUrl = isLiveServer ? "https://api.openai.com/v1" : "http://localhost:9090";
+
+ConnectionConfig config = {
+    auth: {
+        token
+    }
+};
+
+public type DataPassingObject record{
+    string assistantId ="";
+    string threadId = "";
+    string messageId = "";
+    string runId = "";
+    string stepId = "";
+};
+
+DataPassingObject dataPasser = {};
+
+final Client AssistantClient = check new(config,serviceUrl);
+
+const map<string|string[]> headers = {
+    "OpenAI-Beta": ["assistants=v2"]
+};
+
+function dataGen() returns DataPassingObject[][] {
+    return [[dataPasser]];
+}
+
+@test:Config {dataProvider:  dataGen, dependsOn: [testCreateMessage, testCreateThread, testGetMessage]}
+isolated function  testDeleteMessage(DataPassingObject dataPasser) {
+    // Ensure threadId and messageId are not empty
+    if (dataPasser.threadId == "" || dataPasser.messageId == "") {
+        test:assertFail(msg = "No thread ID or message ID available. Ensure message creation test runs first.");
+    }
+
+    // Delete the message
+    DeleteMessageResponse|error res = AssistantClient->/threads/[dataPasser.threadId]/messages/[dataPasser.messageId].delete(headers);
+    if (res is DeleteMessageResponse) {
+        io:println("Message deleted successfully: ", res);
+        test:assertTrue(res.deleted == true, msg = "Failed to delete message");
+    } else {
+        test:assertFail(msg = "Failed to delete message");
+    }
```

Comment:
```suggestion
    DeleteMessageResponse res = check AssistantClient->/threads/[dataPasser.threadId]/messages/[dataPasser.messageId].delete(headers);
    io:println("Message deleted successfully: ", res);
    test:assertTrue(res.deleted == true, msg = "Failed to delete message");
```

Reply:
The resource functions are declared with union types in the client.bal. Is there any way to check the fields of the response?

Reply:
I think this approach fix the problem right ? Shall we do this and see ?